Source/JavaScriptCore/debugger/Debugger.cpp

```diff
@@ -1056,11 +1076,21 @@ void Debugger::exception(JSGlobalObject* globalObject, CallFrame* callFrame, JSV
         setSteppingMode(SteppingModeEnabled);
     }
 
+    // When callFrameMayBeNull is nullptr, we are throwing an error without JS call frames.
+    // This can happen when program throws SyntaxError without evaluation.
+    EmptyTopLevelCallFrameForDebugger emptyCallFrame(globalObject);
```

Comment:
NIT: Could we move this inside the `if` on :1083 so that we only create it when needed, or does this have to be outside?

Reply:
Unfortunately, it does not work since we need to allocate stack space, and we use that stack space in L1088's `updateCallFrame`. We can still make this `std::optional<>` and emplace it inside `if`, but it anyway allocates stack space, and since EmptyTopLevelCallFrameForDebugger is just putting random cells into stack space, it is minimal. So I think just creating `EmptyTopLevelCallFrameForDebugger` here is cleaner.